roles/lsaai/templates/pre_10_setup_connection

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# Pre-file for LDAP service providing authz information.
+LDAP_LOGIN="{{ ldap_domains['stack']['base'] }}"
+LDAP_PASSWORD="{{ ldap_credentials['stack']['admin']['pw'] }}"
+LDAP_URL="ldaps://{% set first = False %}{% for network_id in ip_addresses[inventory_hostname] %}
+      {%- if ip_addresses[inventory_hostname][network_id]['fqdn'] is defined and
+             ip_addresses[inventory_hostname][network_id]['fqdn'] | length >= 1 and
+             ip_addresses[inventory_hostname][network_id]['fqdn'] != 'NXDOMAIN' and
+             not first -%}
+               {{ ip_addresses[inventory_hostname][network_id]['fqdn'] }}
+               {% set first = True %}
+      {%- endif %}
+    {%- endfor %}:636"
```

Comment:
The updated code does not work as it inserts spaces and a newline inside the LDAP_URL declaration. E.g.
```
LDAP_URL="ldaps://some.cool.domain
               :636"
```
Taking the Jinja _for_ and _if_ statements out of the Bash variable declaration and using ```#jinja2: trim_blocks:True, lstrip_blocks: True``` as previously suggested improves readability and is less prone to break IMHO.
This code will also fail with
```
object of type 'NoneType' has no len()
```
when no value is present. E.g. the code handles these cases
```
ip_addresses:
  machine:
    network:
       fqdn: ''
ip_addresses:
  machine:
    network:
       fqdn: ""
```
but it will fail on
```
ip_addresses:
  machine:
    network:
       fqdn:
```

Reply:
Also `{%-` is there to handle empty spaces. Except if you mean that if the fqdn is written in multiple lines :raised_eyebrow: but that should be also covered with the `{%-`.

`| length >= 1` removes the lines that are `''` or `""`.

In short: I tested the code with multiple fqdn's and it worked.

But if this is still insufficient, then I guess that
- `group_vars/jumphost.yml` line 12
- `group_vars/data_transfer.yml` line 10
- `roles/ssh/defaults/main.yml` lines 6 and 16
are also problematic. As they all will also
> fail with
> object of type 'NoneType' has no len()
> when no value is present.

Anyhow: this problematic part has just now being changed to

`LDAP_URL="ldap://localhost"`

As I received the information from LSAAI that
`I could do ldapsearch with this hostname: ldap://localhost. ... Could you please update it in the prescript?`